app/controllers/admin/users_controller.rb

```diff
@@ -49,6 +49,14 @@ def user_status
       redirect_to "/admin/users/#{@user.id}/edit"
     end
 
+    def export_data
+      user = User.find(params[:id])
+      send_to_admin = JSON.parse(params[:send_to_admin])
```

Comment:
Why do you use `JSON.parse` here? I would think the param would be a simple true/false.

Reply:
The form sends the param as a string, and I want to pass along a boolean instead.

Reply:
I did not know JSON.parse would return a boolean for you from a "true"/"false" string. NEAT! TIL!

Reply:
Yeah! Better than trying to a write a custom method IMO :)

Reply:
@Zhao-Andy While `JSON.load` works, it's slightly confusing for the reader. Rails has built-in typecasting that you can use for this (it works with more than just simple "true"/"false", e.g. "on"/"off"):

```ruby
ActiveModel::Type::Boolean.new.cast(params[:send_to_admin])
```

It's a bit more verbose, but also a lot less confusing. Tbh I often wrap this up in a string monkey-patch or utility function:

```class
# somewhere, e.g. `app/refinements`:
module StringRefinements
  refine String do
    def to_bool
      ActiveModel::Type::Boolean.new.cast(self)
    end
  end
end

# Then
"false".to_bool
NoMethodError: undefined method `to_bool' for "false":String
using StringRefinements
"false".to_bool
#=> false
```

Reply:
There's of course also the good old `send_to_admin = params[:send_to_admin] == "true"` approach. In this specific case, I think having `to_bool` available could be worse the extra effort since based on past experience there are different hacks used in different places and standardizing one doesn't hurt.

Reply:
🤯 🤯 🤯 wow, capital TIL!!! I'll add the refinement -- I definitely think we will have cases where we'll use `.to_bool`.

Reply:
TIL 🤯